src/Components/Aspire.Confluent.Kafka/AspireKafkaConsumerExtensions.cs

```diff
@@ -0,0 +1,175 @@
+    /// <param name="configureSettings">An optional method for customizing the <see cref="KafkaConsumerSettings"/>.</param>
+    /// <param name="configureBuilder">An optional method used for customizing the <see cref="ConsumerBuilder{TKey,TValue}"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire:Kafka:Consumer:{name}" section.</remarks>
+    public static void AddKeyedKafkaConsumer<TKey, TValue>(this IHostApplicationBuilder builder, string name, Action<KafkaConsumerSettings>? configureSettings = null, Action<ConsumerBuilder<TKey, TValue>>? configureBuilder = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        AddKafkaConsumer(builder, $"{DefaultConfigSectionName}:{name}", configureSettings, configureBuilder, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddKafkaConsumer<TKey, TValue>(
+        IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<KafkaConsumerSettings>? configureSettings,
+        Action<ConsumerBuilder<TKey, TValue>>? configureBuilder,
+        string connectionName,
+        string? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = BuildConsumerSettings(builder, configurationSectionName, configureSettings, connectionName);
+
+        if (serviceKey is null)
+        {
+            builder.Services.AddSingleton<ConsumerConnectionFactory<TKey, TValue>>(sp => CreateConsumerConnectionFactory<TKey, TValue>(sp, configureBuilder, settings));
+            builder.Services.AddSingleton<IConsumer<TKey, TValue>>(sp => sp.GetRequiredService<ConsumerConnectionFactory<TKey, TValue>>().Create());
+        }
+        else
+        {
+            builder.Services.AddKeyedSingleton<ConsumerConnectionFactory<TKey, TValue>>(serviceKey, (sp, key) => CreateConsumerConnectionFactory<TKey, TValue>(sp, configureBuilder, settings));
+            builder.Services.AddKeyedSingleton<IConsumer<TKey, TValue>>(serviceKey, (sp, key) => sp.GetRequiredKeyedService<ConsumerConnectionFactory<TKey, TValue>>(key).Create());
+        }
+
+        if (settings.Metrics)
+        {
+            builder.Services.TryAddSingleton<MetricsChannel>();
+            builder.Services.AddHostedService<MetricsService>();
+            builder.Services.TryAddSingleton<ConfluentKafkaMetrics>();
+            builder.Services.AddOpenTelemetry().WithMetrics(metricBuilderProvider => metricBuilderProvider.AddMeter(ConfluentKafkaCommon.MeterName));
+        }
+
+        if (settings.HealthChecks)
+        {
+            string healthCheckName = serviceKey is null
+                ? ConfluentKafkaCommon.ConsumerHealthCheckName
+                : string.Format(CultureInfo.InvariantCulture, ConfluentKafkaCommon.KeyedConsumerHealthCheckNameFormat, connectionName);
```

Comment:
Could just do a simple concat here instead of string.Format.

Reply:
done